src/plone/restapi/types/adapters.py

```diff
@@ -110,38 +113,57 @@ def get_type(self):
 @adapter(ITextLine, Interface, Interface)
 @implementer(IJsonSchemaProvider)
 class TextLineJsonSchemaProvider(DefaultJsonSchemaProvider):
+    def additional(self):
+        info = {}
+        if self.field.min_length:
+            info["minLength"] = self.field.min_length
+
+        if self.field.max_length:
+            info["maxLength"] = self.field.max_length
+
+        return info
+
     def get_type(self):
         return "string"
 
 
 @adapter(IText, Interface, Interface)
 @implementer(IJsonSchemaProvider)
 class TextJsonSchemaProvider(TextLineJsonSchemaProvider):
-    def additional(self):
-        info = {}
-        if self.field.min_length is not None:
-            info["minLength"] = self.field.min_length
+    def get_widget(self):
+        return "textarea"
 
-        if self.field.max_length is not None:
```

Comment:
@avoinea doesn't "Text" require a min/max length attribute as well or am I missing something here?

Reply:
@tisto This diff is not that accurate, I moved the additional method within the parent class `TextLineJsonSchemaProvider`